<!-- Launch file for the rrt-detector and the assigner -->
<launch>
  <!-- argument need to change for deployment-->
  <arg name="global_map" value ="/map"/>
  <arg name="global_frame" value ="map"/>
  <arg name="global_costmap_topic" value="/move_base_node/global_costmap/costmap"/>
  <arg name="plan_service" value="/move_base/NavfnROS/make_plan"/>
  <arg name="robot_base_link" value="base_link"/>
  <arg name="move_base_service" value="/move_base"/>


  <!-- robot name space definition -->
  <arg name="first_robot"  default="robot_1"/>

  <!-- RRT common parameters -->
  <arg name="eta" value="1.0"/>
  <arg name="Geta" value="1.0"/>
  <arg name="info_radius" value="1.0"/>
  <arg name="info_multiplier" value="3.0"/>
  <arg name="hysteresis_radius" value="3.0"/>
  <arg name="hysteresis_gain" value="2.0"/>
  <arg name="costmap_clearing_threshold" value="60"/>
  <arg name="delay_after_assignement" value="2.8"/>
  <arg name="goal_time_thres" value="12"/>
  <arg name="frequency" value="50"/>
  <arg name="detected_frontier" value="/detected_points"/>
  <arg name="filtered_frontier" value="/filtered_points"/>

  <!-- global RRT detector -->
  <node pkg="rrt_exploration" type="global_rrt_detector" name="global_rrt_detector" output="screen">
    <param name="eta" value="$(arg Geta)"/>
    <param name="map_topic" value="$(arg global_map)"/>
  </node>

  <!-- filtering node -->
  <node pkg="rrt_exploration" type="filter.py" name="filter" output="screen">
    <param name="map_topic" value="$(arg global_map)"/>
    <param name="info_radius" value="$(arg info_radius)"/> 
    <param name="costmap_clearing_threshold" value="$(arg costmap_clearing_threshold)"/> 
    <param name="global_costmap_topic" value="$(arg global_costmap_topic)"/> 
    <param name="goals_topic" value="$(arg detected_frontier)"/> 
    <param name="robot_namelist" value="$(arg first_robot)"/>
    <param name="rate" value="$(arg frequency)"/>
  </node>

  
  <!-- assigner node -->
  <node pkg="rrt_exploration" type="assigner.py" name="assigner" output="screen">
    <param name="map_topic" value="$(arg global_map)"/>
    <param name="global_frame" value="$(arg global_frame)"/>
    <param name="info_radius" value="$(arg info_radius)"/> 
    <param name="info_multiplier" value="$(arg info_multiplier)"/> 
    <param name="hysteresis_radius" value="$(arg hysteresis_radius)"/> 
    <param name="hysteresis_gain" value="$(arg hysteresis_gain)"/> 
    <param name="frontiers_topic" value="$(arg filtered_frontier)"/> 
    <param name="robot_namelist" value="$(arg first_robot)"/>
    <param name="time_per_meter" value="$(arg time_per_meter)"/>
    <param name="delay_after_assignement" value="$(arg delay_after_assignement)"/>
    <param name="rate" value="$(arg frequency)"/>
    <param name="plan_service" value="$(arg plan_service)"/>
    <param name="base_link" value="$(arg robot_base_link)"/>
    <param name="move_base_service" value="$(arg move_base_service)"/>
  </node>

  </launch>
  
  